witchcraft/witchcraft.go

```diff
@@ -601,17 +603,42 @@ func (s *Server) Start() (rErr error) {
 	ctx = metrics.WithRegistry(ctx, metricsRegistry)
 
 	// initialize loggers
-	s.initLoggers(baseInstallCfg.UseConsoleLog, wlog.InfoLevel, metricsRegistry)
-
-	// add loggers to context
-	ctx = s.withLoggers(ctx)
+	s.initLoggers(baseInstallCfg.UseConsoleLog, wlog.InfoLevel, metricsRegistry, nil)
 
 	// load runtime configuration
 	baseRefreshableRuntimeCfg, refreshableRuntimeCfg, configReloadHealthCheckSource, err := s.initRuntimeConfig(ctx)
 	if err != nil {
 		return err
 	}
 
+	// extract the TCP JSON receiver from the runtime config
+	servicesCfg := baseRefreshableRuntimeCfg.CurrentBaseRuntimeConfig().ServiceDiscovery
+	receiverCfg, err := servicesCfg.MustClientConfig("sls-log-tcp-json-receiver")
+	if err == nil && len(receiverCfg.URIs) > 0 {
+		tlsConfig, err := tlsconfig.NewClientConfig(
+			tlsconfig.ClientRootCAFiles(receiverCfg.Security.CAFiles...),
+			tlsconfig.ClientKeyPairFiles(receiverCfg.Security.CertFile, receiverCfg.Security.KeyFile),
+		)
+		if err != nil {
+			return err
```

Comment:
Can we add message/context to error that says that security material could not be created for the TCP JSON receiver?